x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/set.tsx

```diff
@@ -29,19 +30,55 @@ const fieldsConfig: FieldsConfig = {
   // This is a required field, but we exclude validation because we accept empty values as ''
   value: {
     type: FIELD_TYPES.TEXT,
-    deserializer: String,
+    serializer: from.emptyStringToUndefined,
     label: i18n.translate('xpack.ingestPipelines.pipelineEditor.setForm.valueFieldLabel', {
       defaultMessage: 'Value',
     }),
     helpText: (
       <FormattedMessage
         id="xpack.ingestPipelines.pipelineEditor.setForm.valueFieldHelpText"
-        defaultMessage="Value for the field. A blank value will set {emptyString}."
+        defaultMessage="Value for the field. Used only when {field} is not specified."
         values={{
-          emptyString: <EuiCode>{'""'}</EuiCode>,
+          field: <EuiCode>{'copy_from'}</EuiCode>,
         }}
       />
     ),
+    fieldsToValidateOnChange: ['fields.copy_from', 'fields.value'],
+    validations: [
+      {
+        validator: ({ value, path, formData }) => {
+          if (isEmpty(value) && isEmpty(formData['fields.copy_from'])) {
+            return { path, message: 'Either this field or copy_from should be specified.' };
+          }
+        },
+      },
+    ],
+  },
+  copy_from: {
+    type: FIELD_TYPES.TEXT,
+    serializer: from.emptyStringToUndefined,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.setForm.copyFromFieldLabel', {
+      defaultMessage: 'Copy from',
+    }),
+    helpText: (
+      <FormattedMessage
+        id="xpack.ingestPipelines.pipelineEditor.setForm.copyFromFieldHelpText"
+        defaultMessage="The origin field which will be copied to field. Used only when {field} is not specified."
```

Comment:
[nit] I think this should be "The origin field which will be copied to **the** field..." or, keep as-is, but "field" should be wrapped in `EuiCode`.

I'll defer to the docs team though on this 😄.